pkgs/development/tools/parsing/tree-sitter/update_impl.py

```diff
@@ -106,9 +106,23 @@ def fetchRepo() -> None:
             release: str
             match curl_result(out):
                 case "not found":
-                    # github sometimes returns an empty list even tough there are releases
-                    log(f"uh-oh, latest for {orga}/{repo} is not there, using HEAD")
-                    release = "HEAD"
+                    # if there is a branch provided fetch latest commit from that branch
+                    # otherwise fallback to using HEAD
+                    if "branch" in jsonArg:
+                        branch = jsonArg.get("branch")
+                        branch_out = run_cmd(curl_github_args(
+                            token,
+                            url=f"https://api.github.com/repos/{quote(orga)}/{quote(repo)}/branches/{quote(branch)}"
+                        ))
+                        match curl_result(branch_out):
+                            case {"commit": {"sha": commit_hash}}:
+                                release = commit_hash
```

Comment:
I think we can just do `f"refs/heads/{branch}"` here without the extra api request, `nix-prefetch-git` still outputs a sha as `rev` as expected, using the command from `nix_prefetch_git_args`

```console
$ nix-prefetch-git --quiet --no-deepClone --url https://github.com/derekstride/tree-sitter-sql --rev refs/heads/gh-pages
{
  "url": "https://github.com/derekstride/tree-sitter-sql",
  "rev": "721087c8819cda10ca37f974e914ab9be46b290f",
  "date": "2023-05-19T15:56:01+00:00",
  "path": "/nix/store/5mbri1m7rrwmvhzp5nz3bs1pdzj87fni-tree-sitter-sql",
  "sha256": "10njfi84b78z036bkgbrxnxgy9av9yprxq1p2zla87w0g6ixqva7",
  "fetchLFS": false,
  "fetchSubmodules": false,
  "deepClone": false,
  "leaveDotGit": false
}
```

Reply:
Right, great :) this will simplify things.